app/services/recent-events.ts

```diff
@@ -334,7 +391,102 @@ export class RecentEventsService extends StatefulService<IRecentEventsState> {
     return fetch(request).then(handleResponse);
   }
 
+  get filters() {
+    const mainFilters = pick(this.state.filterConfig, [
+      'donation',
+      'merch',
+      'follow',
+      'host',
+      'bits',
+      'raid',
+      'subscriber',
+      'sponsor',
+      'superchat',
+      'sticker',
+      'effect',
+      'facebook_support',
+      'facebook_like',
+      'facebook_share',
+      'facebook_stars',
+    ]);
+
+    const subFilters = pick(this.state.filterConfig, [
+      'subscription',
+      'subscription_tier_1',
+      'subscription_tier_2',
+      'subscription_tier_3',
+      'primesub',
+      'gifted_sub',
+    ]);
+
+    const resubFilters = pick(this.state.filterConfig, [
+      'resub',
+      'resub_tier_1',
+      'resub_tier_2',
+      'resub_tier_3',
+      'resub_prime',
+      'filter_subscription_3_months',
+      'filter_subscription_6_months',
+      'filter_subscription_9_months',
+      'filter_subscription_12_months',
+      'filter_subscription_minimum_enabled',
+      'filter_subscription_minimum_months',
+    ]);
+
+    const main = {};
+    const sub = {};
+    const resub = {};
+
+    Object.keys(mainFilters).forEach(filter => {
+      main[filter] = {
+        value: mainFilters[filter],
+        name: filterName(filter),
+      };
+    });
+
+    Object.keys(subFilters).forEach(filter => {
+      sub[filter] = {
+        value: subFilters[filter],
+        name: filterName(filter),
+      };
+    });
+
+    Object.keys(resubFilters).forEach(filter => {
+      resub[filter] = {
+        value: resubFilters[filter],
+        name: filterName(filter),
+      };
+    });
+
+    return {
+      main,
+      sub,
+      resub,
+    };
+  }
```

Comment:
This function throws away way too much typing information and should be refactored.  Recognize that this API is very dynamic but I think we can do better.  Let's revisit this together after we get this in preview.